analyses/cell-type-wilms-tumor-06/notebook_template/05_copykat_exploration.Rmd

```diff
@@ -0,0 +1,256 @@
+ - Loss of Chr1p
+ - Gain of Chr1q
+ - Loss of Chr11p13
+ - Loss of Chr11p15
+ - Loss of Chr16q
+                                                                                                        
+Although these are the most frequent, there are patients who do not have any of these alterations and patients that only have some of these alterations. 
+See [Tirode et al.,](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4264969/) and [Crompton et al.](https://doi.org/10.1158/2159-8290.CD-13-1037). 
+                                                                                                        
+`CopyKAT` outputs a matrix that contains the estimated copy numbers for each gene in each cell. 
+We can read that in and look at the mean estimated copy numbers for each chromosome across each cell. 
+We might expect that tumor cells would show an increased estimated copy number in Chr1q, and/or a loss of Chr1p, Chr11p and Chr16q.
+
+```{r}
+# read in full gene by cell copy number detection results
+full_ck_results_df <- full_ck_result_paths |>
+    purrr::map(readr::read_tsv) |>
+    dplyr::bind_rows(.id = "reference_used")
+                                                                                                        
+# for every cell, calculate the mean detection level across all genes in a given chromosome
+full_cnv_df <- full_ck_results_df |>
+    tidyr::pivot_longer(
+    cols = -c(
+    reference_used,
+    chrom
+    ),
+    names_to = "barcodes",
+    values_to = "cnv_detection"
+    ) |>
+    dplyr::group_by(chrom, barcodes, reference_used) |>
+    dplyr::summarise(mean_cnv_detection = mean(cnv_detection))
+                                                                                                        
+# join with cnv info
+cnv_df <- cnv_df |>
+   dplyr::left_join(full_cnv_df, by = c("barcodes", "reference_used")) |>
+   dplyr::filter(!is.na(chrom))    
+
+```
+                                                                                                        
+Let's look at the distribution of CNV estimation in cells that are called aneuploid and diploid by `CopyKAT`. 
+
+```{r, fig.height=15, fig.width=10}
+# create faceted density plots showing estimation of CNV detection across each chr of interest
+# colored by aneuploid/diploid estimation
+ggplot(cnv_df, aes(x = mean_cnv_detection, color = copykat.pred)) +
+  geom_density() +
+  theme_bw() +
+  facet_grid(
+    rows = vars(chrom),
+    cols = vars(reference_used)
+  )
+```
+
+
+
+## Conclusions
+
+From the heatmap of CNV and the mean CNV detection plots, it is difficult to see any CNV pattern that drive the identification of aneuploid cells. 
+The assignment of the aneuploidy/diploidy value might relies on very few CNV and/or an arbitrary threshold. 
+This might be why the assignement of aneuploidy/diploidy values differs between condition (and between runs!!).
```

Comment:
yes, I figured this out when running multiple times to save the heatmap not in jpeg but png format. I can send you examples tomorrow (I struggle with the vpn connexion now 😞, hopefully better tomorrow! )

Reply:
@sjspielman some example to illustrate. 
For the sample -208, I show you the heatmap of CNV for `copykat` ran with an euclidean distance, no reference, two days apart (to then save the heatmap in png instead of jpeg:
![image](https://github.com/user-attachments/assets/7f419505-42e1-4f1b-8f5a-8162e13bfec5)
![image](https://github.com/user-attachments/assets/9a879e3a-5260-43f1-ba16-a72da0839da9)

This is not completly different but not identical.

Reply:
for sample -205
![image](https://github.com/user-attachments/assets/ea77c0f9-7097-47f3-a2cd-94f70407164c)
![image](https://github.com/user-attachments/assets/a18ceb88-a3b9-456c-bd23-7bee8c4f15a1)

Reply:
for -184 it is much more striking ❓❗ 
![image](https://github.com/user-attachments/assets/eee3db63-2145-4a9e-92b8-674c9b7f1157)